### DISCUSSION TITLE
HTTP Request to server running in sidecar

### DISCUSSION THREAD

**Discussion by nikolaswise** [tauri-apps/tauri#5391] [Q&A]
I've got a sidecar running [oxigraph_server](https://crates.io/crates/oxigraph_server) (I think it's running at least!) but I cannot find any documentation on how to make HTTP requests to this server from my front-end application. I've found some issues referencing that this is doable, but I cannot find any documentation around sidecars _at all_.

What is going on here? How do I HTTP my server binary?

**Comment by FabianLars** [comment]
It _should_ be the same as any other server (remote). So the webview's built-in fetch/http should work and if it doesn't there is Tauri's own http module: https://tauri.app/v1/api/js/http

**Reply by nikolaswise** [reply]
Oh I see! So it will just run at `localhost:port` and be a regular ol' thing. That means my sidecar server is _not_ running when I start the application. I guess I'm not totally clear on where or how I execute the sample code [provided in the guide](https://tauri.app/v1/guides/building/sidecar) – I have a template generated main rust file:

```
fn main() {
    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![greet])
        .invoke_handler(tauri::generate_handler![sparql_query])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");Z
}
```

Where do I want to run the Command?

```
let (mut rx, mut child) = Command::new_sidecar("oxigraph_server")
  .expect("failed to create `oxigraph_server` binary command")
  .spawn()
  .expect("Failed to spawn sidecar");
```

I'm assuming _that_ is what executes the binary – it's also unclear to be how I run `oxigraph_server serve`; the only example I saw that used `.args(["serve"])` before `.spawn()` did not work out.

**Reply by nikolaswise** [reply]
Oh wait this seems like it works great actually

```
fn main() {
    use tauri::api::process::Command;
    Command::new_sidecar("oxigraph_server")
      .expect("failed to create `oxigraph_server` binary command")
      .args(["serve"])
      .spawn()
      .expect("Failed to spawn sidecar");

    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![greet])
        .invoke_handler(tauri::generate_handler![sparql_query])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

**Reply by FabianLars** [reply]
unrelated to the issue, but you can call invoke_handler only once. Subsequent calls will overwrite the ones before that, in your case it should look like this (even tho you probably don't need the greet function, but in case you'll add other commands later)
```rs
tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![greet, sparql_query])
        ...
```

> Where do I want to run the Command?

Probably in the setup hook, similar to this: https://github.com/tauri-apps/tauri/blob/dev/examples/sidecar/src-tauri/src/main.rs but what you're doing _should_ work fine too yeah

**Reply by nikolaswise** [reply]
🙇 Thank you!
